src/commands/warn.ts

```diff
@@ -0,0 +1,118 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { Command, CommandOptionsRunTypeEnum } from '@sapphire/framework';
+import {
+    PermissionFlagsBits,
+    EmbedBuilder,
+    type ChatInputCommandInteraction,
+} from 'discord.js';
+import { Color } from '../lib/embeds.js';
+import { messageLogger, moderationLogs } from '../index.js';
+import type { warnLog } from '../lib/moderation.js';
+
+@ApplyOptions<Command.Options>({
+    description: 'Warn user',
+    requiredClientPermissions: [PermissionFlagsBits.ModerateMembers],
+    requiredUserPermissions: [PermissionFlagsBits.ModerateMembers],
+    runIn: [CommandOptionsRunTypeEnum.GuildAny],
+})
+export class WarnCommand extends Command {
+
+    public override async chatInputRun(interaction: ChatInputCommandInteraction) {
+        if (!interaction.inGuild()) {
+            return;
+        }
+        const user = interaction.options.getUser(Option.User, true);
+        const reason = interaction.options.getString(Option.Reason);
```

Comment:
The reason is already set to required in the option itself, so we just need to tell Discord.js that it's required when we obtain its value, like we do for user, so that we won't get a "maybe null" value back.
```suggestion
        const reason = interaction.options.getString(Option.Reason, true);
```